src/components/multi-action-button/multi-action-button.component.tsx

```diff
@@ -41,17 +42,21 @@ export const MultiActionButton = ({
 }: MultiActionButtonProps) => {
   const ref = useRef<HTMLDivElement>(null);
   const buttonRef = useRef<HTMLButtonElement>(null);
-  const buttonContainer = useRef<HTMLDivElement>(null);
-  const buttonChildren = useMemo(() => React.Children.toArray(children), [
-    children,
-  ]);
-  const buttonChildrenRefs = useMemo(
-    () => buttonChildren.map(() => React.createRef<HTMLButtonElement>()),
-    [buttonChildren]
+  const childrenContainer = useRef<HTMLDivElement>(null);
+
+  const getButtonChildren = useCallback(
```

Comment:
**question**: as we're doing the same thing in both Multi and Split components, I wonder if it would be worth moving this into a custom hook or a provider component? This is non-blocking for me as it may be more complicated than to integrate it with `useMenuKeyboardNavigation` being used here as well but figured I'd ask anyway 😄

Reply:
Fair comment. I did realise there was a lot of duplication between the two components but realised a lot of it already existed so didn't really explore further - but now you mention it I agree it's worth exploring whether some of this duplication can be put into a custom hook 👍